src/kprofile.cpp

```diff
@@ -18,40 +23,66 @@ KProfile::KProfile(QObject *parent) :
 #endif
 }
 
-#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
-void KProfile::startDownload(QWebEngineDownloadItem* download)
-#else
-void KProfile::startDownload(QWebEngineDownloadRequest* download)
-#endif
+namespace {
+    void setDownloadFilePath(KProfile::WebEngineDownloadType* download, QString filePath)
+    {
+    #if QT_VERSION < QT_VERSION_CHECK(5, 14, 0)
+        download->setPath(filePath);
+    #else
+        download->setDownloadFileName(filePath);
+    #endif
+    }
+}
+
+void KProfile::openFile(WebEngineDownloadType* download)
+{
+    QString defaultFileName = download->url().fileName();
+    QTemporaryFile tempFile(QDir::tempPath() + "/XXXXXX." + QFileInfo(defaultFileName).suffix());
+    tempFile.setAutoRemove(false);
+    if (tempFile.open()) {
+        QString tempFilePath = tempFile.fileName();
+        tempFile.close();
+        setDownloadFilePath(download, tempFilePath);
+        connect(download, &WebEngineDownloadType::finished, [tempFilePath]() {
+            if(!QDesktopServices::openUrl(QUrl::fromLocalFile(tempFilePath)))
+                showInfoBox(gt("error-title"), gt("error-opening-file"), KiwixApp::instance()->getMainWindow());
+        });
+        download->accept();
+    } else {
+        qDebug()<<"tmp file err";
+        download->cancel();
+    }
+}
+
+void KProfile::saveFile(WebEngineDownloadType* download)
 {
     QString defaultFileName = download->url().fileName();
-    QString fileName = QFileDialog::getSaveFileName(KiwixApp::instance()->getMainWindow(),
-                                                       gt("save-file-as-window-title"), defaultFileName);
+    QString fileName = QFileDialog::getSaveFileName(KiwixApp::instance()->getMainWindow(), "Save File", defaultFileName);
     if (fileName.isEmpty()) {
+        download->cancel();
         return;
     }
     QString extension = "." + download->url().url().section('.', -1);
     if (!fileName.endsWith(extension)) {
         fileName.append(extension);
     }
-#if QT_VERSION < QT_VERSION_CHECK(5, 15, 0)
-    download->setPath(fileName);
-#else
-    download->setDownloadFileName(fileName);
-#endif
-#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
-    connect(download, &QWebEngineDownloadItem::finished, this, &KProfile::downloadFinished);
-#else
-    connect(download, &QWebEngineDownloadRequest::isFinished, this, &KProfile::downloadFinished);
-#endif
```

Comment:
Did you eliminate `KProfile::downloadFinished()` intentionally? Now when the file is saved no `gt("download-finished-message")` message will be shown to the user.

Reply:
Yes, it was intentional. I forgot to add the download finished message in saveFile().